### PR TITLE
[Fix](MOW) Fix load data publish timeout when enable unique key MOW

### LINKED ISSUE #20719: [Bug] load data publish timeout when use unique key MOW
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0 beta

### What's Wrong?

"Comment": "",
"TwoPhaseCommit": "false",
"Status": "Publish Timeout",
"Message": "[PUBLISH_TIMEOUT]transaction commit successfully, BUT data will be visible later",
"NumberTotalRows": 2648836,
"NumberLoadedRows": 2648834,
"NumberFilteredRows": 2,
"NumberUnselectedRows": 0,
"LoadBytes": 187645536,
"LoadTimeMs": 15815,
"BeginTxnTimeMs": 0,
"StreamLoadPutTimeMs": 1,
"ReadDataTimeMs": 102,
"WriteDataTimeMs": 10811,
"CommitAndPublishTimeMs": 0,

### What You Expected?

no publish time out

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

Issue Number: close #20719

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run p0

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

File: be/src/olap/tablet.cpp [hunk 1]
```diff
@@ -3126,10 +3126,34 @@ Status Tablet::update_delete_bitmap_without_lock(const RowsetSharedPtr& rowset)
...
-    const RowsetIdUnorderedSet& pre_rowset_ids = load_info->rowset_ids;
+Status Tablet::commit_phase_update_delete_bitmap(
+        const RowsetSharedPtr& rowset, const RowsetIdUnorderedSet& pre_rowset_ids,
+        DeleteBitmapPtr delete_bitmap, const int64_t& cur_version,
+        const std::vector<segment_v2::SegmentSharedPtr>& segments, RowsetWriter* rowset_writer) {
+    RowsetIdUnorderedSet cur_rowset_ids;
+    RowsetIdUnorderedSet rowset_ids_to_add;
+    RowsetIdUnorderedSet rowset_ids_to_del;
+
+    std::lock_guard<std::mutex> rwlock(_rowset_update_lock);
```
Comment (liaoxin01) [inline be/src/olap/tablet.cpp:3137]: don't need  to lock _rowset_update_lock here?

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run p0

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run p0

**Pull request merged**